### ISSUE TITLE
Bug - Field values are being copied if you duplicate a pre-filled dynamic area

### ISSUE THREAD

**Issue by kaphert** [neverwoods/validformbuilder#59]
If you prefill fields in a dynamic area and you add a new dynamic area by pressing on '+ add' the values of the first dynamic area are being copied to the new dynamic area. 

Example code 
```php
$form = new ValidForm('formName');

$form->addField('simple', 'Simple field',ValidForm::VFORM_STRING);

//*** Output submitted data or the form

$area = $form->addArea(
    "Dynamic area",
    false,
    "dyn-area",
    false,
    [
        "dynamic" => true,
        "dynamicLabel" => "+ add"
    ]
);

$area->addField(
    "field-a",
    "Field A",
    ValidForm::VFORM_STRING,
    [
        "required" => true
    ],
    [
        "required" => "This field is required"
    ]
);

$area->addField(
    "field-b",
    "Field B",
    ValidForm::VFORM_STRING
);

$form->setDefaults([
    'simple' => 'Simple value',
    'field-a' => ['Field A 1', 'Field A 2'],
    'field-b' => ['Field B 1', 'Field B 2'],
]);

if ($form->isSubmitted() && $form->isValid()) {
    echo $form->valuesAsHtml();
} else {
    echo $form->toHtml();
}
```

**Comment by rvanbaalen** [comment]
Currently, this is expected behaviour. However I would argue that this is indeed unwanted and should change.

What are your two cents @flangfeldt? If you agree with me, you can look into my Pull Request to change this behavior:

**Comment by flangfeldt** [comment]
Which PR is that?

The point of having a "default" value is so that one can force the state of the field. Right now the "default" property is almost exclusively used to set a previously saved value. These are 2 different use cases in my opinion.

Imagine having a drop down field with 5 values. You want one of the values to be selected "by default", meaning if the user does nothing, this is the value the field should have. If this field is a dynamic field you probably want the value of the duplicated field to be the preset "default" value.

Now imagine you have the same dropdown field and it is filled with a previously saved value using the "default" property. Now, when the user duplicates the field you probably don't want the new field to inherit the "default" property (saved value), but actually use the "default" value that was set earlier by the business logic.

See where this is going?

**Comment by rvanbaalen** [comment]
The PR was a work in progress; didn't expect a response so fast ;-) haha. 

Anyway, I see exactly where this is going. The current situation shouldn't change perhaps. In an ideal situation there should be two ways to set a value: define a 'default' value and/or define a 'previous'/'preset' value?

So this isn't a bug rather than a feature suggestion

**Comment by flangfeldt** [comment]
Yes, it's a feature request. The current behaviour is expected.
Implementing 2 properties would make thinks a lot easier on the validation part as well. Let's link this to the 5.0 WIP.

**Comment by rvanbaalen** [comment]
Closed in favour of #67
